src/complexity_measures/missing_dispersion.jl

```diff
@@ -0,0 +1,73 @@
+using DelayEmbeddings
+export MissingDispersionPatterns
+
+"""
+    MissingDispersionPatterns <: ComplexityMeasure
+    MissingDispersionPatterns(; est = Dispersion())
+
+An estimator for the number of missing dispersion patterns (``N_{MDP}``), a complexity
+measure which can be used to detect nonlinearity in time series (Zhou et al.,
+2022)[^Zhou2022], using the dispersion pattern estimator `c`.
+
+Used with [`complexity`](@ref) or [`complexity_normalized](@ref)`.
+
+## Description
+
+``N_{MDP}`` is computed by first symbolising each `xᵢ ∈ x`, then embedding the resulting
+symbol sequence (using `est`), and computing the quantity
+
+```math
+N_{MDP} = L - N_{ODP},
+```
+
+where `L = alphabet_length(est)` (i.e. the total number of possible dispersion patterns),
+and ``N_{ODP}`` is defined as the number of *occurring* dispersion patterns.
+
+The normalized complexity measure ``N_{MDP}^N = (L - N_{ODP})/L`` is
+computed. The authors recommend that
+`alphabet_length(est.symbolization)^est.m << length(x) - est.m*est.τ + 1`` to avoid
+undersampling.
+
+## Usage
+
+In Zhou et al. (2022), [`MissingDispersionPatterns`](@ref) is used to detect nonlinearity
+in time series by comparing the ``N_{MDP}`` for a time series `x` to ``N_{MDP}`` values for
+an ensemble of surrogates of `x`. If ``N_{MDP} > q_{MDP}^{WIAAFT}``, where
+``q_{MDP}^{WIAAFT}`` is some `q`-th quantile of the surrogate ensemble, then it is
+taken as evidence for nonlinearity.
+
+See also: [`Dispersion`](@ref), [`ReverseDispersion`](@ref), [`alphabet_length`](@ref).
+
+[^Zhou2022]: Zhou, Q., Shang, P., & Zhang, B. (2022). Using missing dispersion patterns
+    to detect determinism and nonlinearity in time series data. Nonlinear Dynamics, 1-20.
+"""
+struct MissingDispersionPatterns{D} <: ComplexityMeasure
+    est::D
+
+    function MissingDispersionPatterns(est::D) where {D <: Union{Dispersion, ReverseDispersion}}
+        new{D}(est)
+    end
+end
+
+function count_nonoccurring(x::AbstractVector{T}, est) where T
+    τ, m = est.τ, est.m
+
+    # Symbolize and embed
+    symbols = symbolize_for_dispersion(x, est)
+    embedding = genembed(symbols, 0:τ:(m - 1)*τ)
+    n_occuring = length(Set(embedding.data))
+
+    L = alphabet_length(est)
+    n_not_occurring = L - n_occuring
+
+    return L, n_not_occurring
+end
+
+function complexity(c::MissingDispersionPatterns, x::AbstractVector{T}) where T
+    return count_nonoccurring(x, c.est)[2]
+end
```

Comment:
I don't get this source code. The source code, given the docstring, should be:
```julia
probs = probabilities(x, c.est)
L = alphabet_length(x, est)
O = count(!iszero, probs) # correct way to count occurring probabilities for any estimator, in case some estimators also include 0 probabilities
N  = (L-O)/L # actually normalize ? 
return N
```
It is estimator independent and simpler. What am I missing? Reading the docstring that's what we should be doing...?

Reply:
What's even the purpose of `symbolize_for_dispersion`

Reply:
> It is estimator independent and simpler. What am I missing? Reading the docstring that's what we should be doing...?

Your solution is much more elegant. I updated the PR.

> What's even the purpose of symbolize_for_dispersion?

`symbolize_for_dispersion` does a necessary pre-processing step which ensures that the algorithm doesn't crash when there's only a single unique value in the input data.